webui/react/src/pages/F_ExpList/glide-table/custom-renderers/cells/loadingCell.tsx

```diff
@@ -0,0 +1,81 @@
+
+import { Theme } from 'shared/themes';
+import { rgba2str, rgbaFromGradient, str2rgba } from 'shared/utils/color';
+
+import { roundedRect } from '../utils';
+
+interface LoadingCellProps {
+  readonly appTheme: Theme;
+  readonly kind: 'loading-cell';
+}
+
+export type LoadingCell = CustomCell<LoadingCellProps>;
+
+const PADDING = 10;
+const MAX_STEPS = 10;
+
+const renderer: CustomRenderer<LoadingCell> = {
+  draw: (args, cell) => {
+    const { ctx, rect, requestAnimationFrame } = args;
+    const { appTheme } = cell.data;
+
+    ctx.save();
+
+    const x = rect.x + PADDING;
+    const y = rect.y + PADDING;
+    const w = rect.width - 2 * PADDING;
+    const h = rect.height - 2 * PADDING;
+
+    /**
+     * Chart to help figure out all the necessary interval steps to calculate final gradient
+     * colors and color stops.
+     *
+     * percent  inc   factor  color1-stop   color0                color1                color2
+     * -----------------------------------------------------------------------------------------------------
+     * 0.0      0.0   0       0.0           g(rgba0, rgba1, 0.0)  g(rgba0, rgba1, 0.0)  g(rgba0, rgba1, 1.0)
+     * 0.1      0.2   0       0.2           g(rgba0, rgba1, 0.2)  g(rgba0, rgba1, 0.0)  g(rgba0, rgba1, 0.8)
+     * 0.25     0.5   0       0.5           g(rgba0, rgba1, 0.5)  g(rgba0, rgba1, 0.0)  g(rgba0, rgba1, 0.5)
+     * 0.4      0.8   0       0.8           g(rgba0, rgba1, 0.8)  g(rgba0, rgba1, 0.0)  g(rgba0, rgba1, 0.2)
+     * 0.5      0.0   1       0.0           g(rgba0, rgba1, 1.0)  g(rgba0, rgba1, 1.0)  g(rgba0, rgba1, 0.0)
+     * 0.6      0.2   1       0.2           g(rgba0, rgba1, 0.8)  g(rgba0, rgba1, 1.0)  g(rgba0, rgba1, 0.2)
+     * 0.75     0.5   1       0.5           g(rgba0, rgba1, 0.5)  g(rgba0, rgba1, 1.0)  g(rgba0, rgba1, 0.5)
+     * 0.9      0.8   1       0.8           g(rgba0, rgba1, 0.2)  g(rgba0, rgba1, 1.0)  g(rgba0, rgba1, 0.8)
+     */
+
+    const step = Math.round(Date.now() / 100) % MAX_STEPS;
+    const percent = step / MAX_STEPS;
+    const inc = (2 * percent) % 1;
+    const factor = percent >= 0.5 ? 1.0 : 0.0;
+    const rgba0 = str2rgba(appTheme.stage);
+    const rgba1 = str2rgba(appTheme.stageStrong);
+    const color0 = rgba2str(rgbaFromGradient(rgba0, rgba1, Math.abs(factor - inc)));
+    const color1 = rgba2str(rgbaFromGradient(rgba0, rgba1, factor));
+    const color2 = rgba2str(rgbaFromGradient(rgba0, rgba1, 1 - Math.abs(factor - inc)));
+
+    const gradient = ctx.createLinearGradient(x, y, x + w, y);
+    gradient.addColorStop(0.0, color0);
+    gradient.addColorStop((2 * percent) % 1, color1);
+    gradient.addColorStop(1.0, color2);
+
+    ctx.beginPath();
+    roundedRect(ctx, x, y, w, h, 2);
+    ctx.fillStyle = gradient;
+    ctx.fill();
+
+    ctx.restore();
+
+    requestAnimationFrame();
+
+    return true;
+  },
+  isMatch: (cell: CustomCell): cell is LoadingCell =>
+    (cell.data as LoadingCellProps).kind === 'loading-cell',
```

Comment:
nit: its just a preference thing, but i personally prefer  not to use `as` as much as possible.
alternative would be like this. what do you think? 
```ts
return 'kind' in cell.data ? cell.data.kind === 'loading-cell' : false;
``` 

---

also, instead of hard code `loading-cell`, that would be nice to add 

```ts
const LoadingCellKind = {
   LoadingCell:  'loading-cell'
} as const;
```

Reply:
I agree with you with both points of what you said. I'll update the `isMatch`. As for the kind, would it be better to consolidate all the different kinds we have into something like a const dictionary? I think it might be best to do that as a refactor.